implants/lib/transport/src/dns.rs

```diff
@@ -0,0 +1,1627 @@
+                    }
+                }
+            } else if rtype == A_RECORD_TYPE || rtype == AAAA_RECORD_TYPE {
+                // A or AAAA record - extract IP address bytes
+                if offset + rdlength as usize <= response.len() {
+                    record_data.extend_from_slice(&response[offset..offset + rdlength as usize]);
+                    offset += rdlength as usize;
+                }
+            } else {
+                offset += rdlength as usize;
+            }
+        }
+
+        Ok(record_data)
+    }
+
+    /// Send a single DNS query and receive response, with record type fallback
+    async fn send_query(&mut self, subdomain: &str) -> Result<Vec<u8>> {
+        use rand::Rng;
+
+        let socket = self
+            .socket
+            .as_ref()
+            .ok_or_else(|| anyhow::anyhow!("Socket not initialized"))?;
+
+        // Determine which record types to try
+        let record_types_to_try: Vec<u16> = if self.enable_fallback {
+            // Try all record types in priority order, but start with preferred
+            let mut types = Vec::new();
+            types.push(self.preferred_record_type);
+            for &rt in RECORD_TYPE_PRIORITY {
+                if rt != self.preferred_record_type {
+                    types.push(rt);
+                }
+            }
+            types
+        } else {
+            // Only try the preferred record type
+            vec![self.preferred_record_type]
+        };
+
+        // Try each record type
+        for &record_type in &record_types_to_try {
+            #[cfg(debug_assertions)]
+            {
+                let type_name = match record_type {
+                    TXT_RECORD_TYPE => "TXT",
+                    A_RECORD_TYPE => "A",
+                    AAAA_RECORD_TYPE => "AAAA",
+                    _ => "UNKNOWN",
+                };
+                log::trace!("Attempting DNS query with record type: {}", type_name);
+            }
+
+            // Generate random transaction ID
+            let transaction_id: u16 = rand::thread_rng().gen();
+            let query = self.build_dns_query(subdomain, transaction_id, record_type);
+
+            // Determine DNS server to use
+            let target = if let Some(ref server) = self.dns_server {
+                server.clone()
+            } else {
+                // Use system resolver - send to localhost:53
+                "127.0.0.1:53".to_string()
+            };
+
+            // Send query
+            match socket.send_to(&query, &target).await {
+                Ok(_) => {}
+                Err(e) => {
+                    #[cfg(debug_assertions)]
+                    log::trace!("Failed to send query: {}", e);
+                    continue; // Try next record type
+                }
+            }
+
+            // Receive response(s) until we get one with matching transaction ID
+            let deadline = tokio::time::Instant::now() + tokio::time::Duration::from_secs(5);
```

Comment:
Add a constant at the top of the file to define the deadline just makes it easier if we ever need to update it.